### ISSUE TITLE
Use case for running mumble-server with system-wide D-Bus service

### ISSUE THREAD

**Issue by mgerstner** [mumble-voip/mumble#6098]
### The issue

I am currently reviewing a version update of the mumble package in openSUSE to version 1.5.517.

In this version the D-Bus configuration file `/usr/share/dbus-1/system.d/mumble-server.conf` is now installed by default. It seems this configuration will never work, since it is based on user `mumble-server`:

```
  <policy user="mumble-server">
    <allow own="net.sourceforge.mumble.murmur"/>
  </policy>
```

while the systemd-sysusers configuration `/usr/lib/sysusers.d/mumble-server.conf` is for `_mumble-server`:

```
# Type # Name         # ID # GECOS              # Home # Shell
u      _mumble-server -    "Mumble server user" -      -
```

But more important to me is the question what the use case for running this service on the D-Bus system bus is. In the default configuration file for mumble-server the D-Bus service is disabled completely, otherwise a preset for running it on the session bus is suggested.

The D-Bus interface does not implement authentication, so running it on the system bus would mean that any local users with access to the system bus can abuse the interface in various manners.

Can you shed some light onto this?

### Mumble version

1.5.517

### Mumble component

Server

### OS

Linux

**Comment by Krzmbrzl** [comment]
This is essentially the effect of me not really knowing what exactly this config file is and thus simply installing it to where Google told me this kind of stuff is supposed to be installed to.
From your description it seems like it might make more sense to not install it at all. But in that case, we should probably just get rid of it entirely as I don't want to keep unused files in this repo (those only get old and start stinking) :thinking:

**Comment by Krzmbrzl** [comment]
> I am currently reviewing a version update of the mumble package in openSUSE to version 1.5.517.

As an aside: you are aware that 1.5.517 is technically not an actual stable Mumble release, right? I mean on Linux it should probably work just fine, but I wanted to make sure that you know :)

**Comment by mgerstner** [comment]
> > I am currently reviewing a version update of the mumble package in openSUSE to version 1.5.517.
> 
> As an aside: you are aware that 1.5.517 is technically not an actual stable Mumble release, right? I mean on Linux it should probably work just fine, but I wanted to make sure that you know :)

I did not package this update, but I will forward the information to our package maintainer. It's for openSUSE Tumbleweed which is rolling release, but not necessarily intended for unstable versions.

**Comment by mgerstner** [comment]
> This is essentially the effect of me not really knowing what exactly this config file is and thus simply installing it to where Google told me this kind of stuff is supposed to be installed to. From your description it seems like it might make more sense to not install it at all. But in that case, we should probably just get rid of it entirely as I don't want to keep unused files in this repo (those only get old and start stinking) thinking

As a short summary about D-Bus:

There is the system and the session bus. The session bus runs in user context with user privileges once a graphical login happens. Desktop environments like KDE and Gnome run their services like desktop search / whatever services on the session bus. These are more or less just regular applications that as daemons in the background using the well defined D-Bus protocol. Only the user that started the session bus can access it (usually).

The system bus is system wide and usually services running as root or as unprivileged but separate service user accounts are running there. Typicall all other users in the system can make use of these system-wide service. `firewalld` would be an example which allows users to interact with the system's firewall settings.

For running on the session bus no special configuration files should be necessary, since the applications running there are run by the user that owns the bus.

For running on the system bus a configuration file like the one this issue is about is necessary. It explains to D-Bus who may register ("own") a specific D-Bus interface (i.e. start such a D-Bus service) and who will be allowed to talk to it (most of the time all users). For more fine grained authorization checks typicall "Polkit" is used to authenticate fine grained actions that a system service performs.

So long story short: If mumble is not supposed to run as a system service accessible to all users in the system then the above configuration file will not be necessary. If it *is* supposed to be run this way in some context then it will be problematic security wise, because just about any user in the system (even accounts like `nobody`) can then play around with the Mumble settings. That is unless Mumble would implement something like Polkit authentication on top of D-Bus.

**Comment by Krzmbrzl** [comment]
Thank you very much for bringing this to our/my attention and for the explanation. I will remove the DBus config file from our source tree - the DBus interface to the server is considered deprecated anyway.
This change will also be backported to the to-be-released (as a stable release) 1.5 series.
